gadget/layout.go

```diff
@@ -314,9 +314,15 @@ func resolveVolumeContent(gadgetRootDir, kernelRootDir string, kernelInfo *kerne
 // content and populates VolumeContent.resolvedSource with absolute
 // paths.
 func resolveContentPathOrRef(gadgetRootDir, kernelRootDir string, kernelInfo *kernel.Info, pathOrRef string) (string, error) {
-	// TODO: what to here? we could error instead?
-	if pathOrRef == "" {
-		return "", nil
+
+	// TODO: add kernelRootDir == "" error too once all the higher
```

Comment:
Shouldn't this check be under `if strings.HasPrefix(pathOrRef, "$kernel:") {` at that point?

Reply:
no, it's  a programming error not to pass the kernel even there are no $kernel usages. It's no a dynamic problem